website/docs/guides/version_5_upgrade.html.markdown

```diff
@@ -0,0 +1,161 @@
+---
+layout: "vault"
+page_title: "Terraform Vault Provider 5.X Upgrade Guide"
+sidebar_current: "docs-vault-provider-version-5-upgrade"
+description: |-
+  Terraform Vault Provider 5.X Upgrade Guide
+
+---
+
+# Terraform Vault Provider 5.X Upgrade Guide
+
+Version `5.X` of the Vault provider for Terraform is a major release and
+includes some changes that you will need to consider when upgrading. This guide
+is intended to help with that process and focuses only on the changes necessary
+to upgrade from version `4.X` to `5.X`.
+
+Most of the changes outlined in this guide have been previously marked as
+deprecated in the Terraform `plan`/`apply` output throughout previous provider
+releases, up to and including 4.8.0. These changes, such as deprecation notices,
+can always be found in the [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md).
+
+-> If you are upgrading from `1.9.x`. Please follow the
+[2.0.0 Upgrade Guide](./version_2_upgrade.html) before proceeding any further.
+
+-> If you are upgrading from `2.24.X`. Please follow the
+[3.0.0 Upgrade Guide](./version_3_upgrade.html) before proceeding any further.
+
+-> If you are upgrading from `3.25.X`. Please follow the
+[4.0.0 Upgrade Guide](./version_4_upgrade.html) before proceeding any further.
+
+## Why version 5.X?
+
+We introduced version `5.X` in order to multiplex the provider
+to use the [Terraform Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework),
+upgrade to Terraform `1.11.x`, and support Ephemeral Resources and Write-Only attributes.
+The change was deemed significant enough to warrant the major version bump.
+In addition to the aforementioned Framework upgrade, previously deprecated fields have been removed.
+
+While you may see some small changes in your configurations as a result of
+these changes, we don't expect you'll need to make any major refactorings.
+
+## What Vault server versions are supported in version 5.X?
+
+The Vault provider will be dropping Vault version support for Vault <= `1.13.x`.
+This means that only Vault server version `1.14.x` and greater will be supported.
```

Comment:
Confirm if dropping legacy version support is appropriate

Reply:
will be good to confirm this